Jetsnack/app/src/main/java/com/example/jetsnack/ui/home/cart/SwipeDismissItem.kt

```diff
@@ -16,52 +16,46 @@
 
 package com.example.jetsnack.ui.home.cart
 
-import androidx.compose.animation.AnimatedVisibility
 import androidx.compose.animation.EnterTransition
 import androidx.compose.animation.ExitTransition
 import androidx.compose.animation.expandVertically
 import androidx.compose.animation.shrinkVertically
-import androidx.compose.material.DismissDirection
-import androidx.compose.material.ExperimentalMaterialApi
-import androidx.compose.material.SwipeToDismiss
-import androidx.compose.material.rememberDismissState
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.SwipeToDismissBox
+import androidx.compose.material3.rememberSwipeToDismissBoxState
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.platform.LocalDensity
-import androidx.compose.ui.unit.Dp
 
-@OptIn(ExperimentalMaterialApi::class)
+@OptIn(ExperimentalMaterial3Api::class)
 @Composable
 /**
  * Holds the Swipe to dismiss composable, its animation and the current state
  */
 fun SwipeDismissItem(
     modifier: Modifier = Modifier,
-    directions: Set<DismissDirection> = setOf(DismissDirection.EndToStart),
     enter: EnterTransition = expandVertically(),
     exit: ExitTransition = shrinkVertically(),
-    background: @Composable (offset: Dp) -> Unit,
-    content: @Composable (isDismissed: Boolean) -> Unit,
+    background: @Composable (/*offset: Dp*/) -> Unit,
+    content: @Composable (/*isDismissed: Boolean*/) -> Unit,
 ) {
     // Hold the current state from the Swipe to Dismiss composable
-    val dismissState = rememberDismissState()
+    val dismissState = rememberSwipeToDismissBoxState()
     // Boolean value used for hiding the item if the current state is dismissed
-    val isDismissed = dismissState.isDismissed(DismissDirection.EndToStart)
+    // val isDismissed = dismissState.currentValue == SwipeToDismissBoxValue.
     // Returns the swiped value in dp
-    val offset = with(LocalDensity.current) { dismissState.offset.value.toDp() }
+    // val offset = with(LocalDensity.current) { dismissState.offset.value.toDp() }
 
-    AnimatedVisibility(
+    /* AnimatedVisibility(
+         modifier = modifier,
+         visible = !isDismissed,
+         enter = enter,
+         exit = exit
+     ) {*/
+    SwipeToDismissBox(
         modifier = modifier,
-        visible = !isDismissed,
-        enter = enter,
-        exit = exit
-    ) {
-        SwipeToDismiss(
-            modifier = modifier,
-            state = dismissState,
-            directions = directions,
-            background = { background(offset) },
-            dismissContent = { content(isDismissed) }
-        )
-    }
+        state = dismissState,
+        backgroundContent = { background(/*offset*/) },
+        content = { content(/*isDismissed*/) }
+    )
+    /*   }*/
```

Comment:
Remove the commented out sections?

Reply:
Thanks for reminding me about this one, I added it back based on new dismissed state information (progress percentage instead of DP)